tests/instruments/hp/test_hp437b_with_device.py

```diff
@@ -0,0 +1,220 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import pytest
+from time import sleep
+from pymeasure.instruments.hp import HP437B
+from pymeasure.instruments.hp.hp437b import LogLin, MeasurementUnit, OperatingMode, TriggerMode, \
+    GroupTriggerMode
+import numpy as np
+
+pytest.skip('Only work with connected hardware', allow_module_level=True)
+
+
+class TestHP437B:
+    """
+    Unit tests for HP437B class.
+
+    This test suite, needs the following setup to work properly:
+        - A HP437B device should be connected to a gpib capable adapter;
+        - The device's address must be set in the RESOURCE constant.
+        - Some power sensor connected to the power reference reading around 0 dBm (+/- 1 dBm)
+    """
+
+    RESOURCE = 'USB0::0x03EB::0x2065::HEWLETT-PACKARD_437B__2.0::0::INSTR'
+
+    instr = HP437B(RESOURCE)
```

Comment:
See the [documentation](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/tests.html#device-tests) regarding how to give a device address to the test environment.
This additional property will also ensure, that the tests are skipped without a connected device.